src/main/java/com/hms/flexyosisoftconnector/OSIsoftConnectorMain.java

```diff
@@ -72,6 +72,14 @@ public static void main(String[] args) throws IOException, JSONException {
     // update tag info to use later
     TagInfoManager.refreshTagList();
 
+    if (TagInfoManager.getTagInfoList().size() == 0) {
+      Logger.LOG_SERIOUS("There are 0 tags monitored for telemetry data to OSIsoft.");
+      Logger.LOG_SERIOUS(
+          "Add 1 or more tags to telemetry data and restart the connector. Connector application"
+              + " exiting.");
+      System.exit(1);

```

Comment:
You may want to look into adding a connectorShutdown(int exitCode) method to main.
That way anything needing to exit can call that and get a clean exit/shutdown without having a bunch of 
"Connector application exiting." strings thrown throughout the code.

Reply:
made a new issue and will address in next PR